### PR TITLE
[BROOKLYN-542] add test for freemarker builtins hashes

### PR DESCRIPTION
**Pull request by andreaturli**

### CODE REVIEW

**Comment by aledsage** [comment]
Retest this please

File: core/src/test/java/org/apache/brooklyn/util/core/text/TemplateProcessorTest.java [hunk 1]
```diff
@@ -247,4 +251,40 @@ public void testDotSeparatedKeyCollisionFailure() {
...
+        entity.sensors().set(sensor1, ImmutableMap.of("name", "mouse", "price", 50));
+        String keys = Joiner.on("\n").join(
+                "<#list attribute['map']?keys as k>",
+                "${k}",
+                "</#list>");
+        String resultForKeys = TemplateProcessor.processTemplateContents(keys, entity, ImmutableMap.of());
+        assertEquals(resultForKeys.trim(), "name\nprice");
+
+        String values = Joiner.on("\n").join(
+                "<#list attribute['map']?keys as k>",
```
Comment (aledsage) [inline core/src/test/java/org/apache/brooklyn/util/core/text/TemplateProcessorTest.java]: I think this should say `?values...`

**Comment by andreaturli** [comment]
yes @aledsage this fails because of that jira issue.

it's only the first commit to demonstrate the problem. I'll add a fix as soon as I have it. ( still trying to understand freemarker API)

**Comment by andreaturli** [comment]
Close this for long inactivity - thanks @aledsage for your help

**Pull request closed without merging**